scripts/rebuild-endpoints.py

```diff
@@ -39,240 +30,83 @@
-        # special case behavior with services like iam, cloudfront.
-        return (
-            'partitionEndpoint' in
-            self._endpoint_data['partitions'][0]['services'].get(
-                service_name, {}))
 
-
-class StaticEndpointBuilder(object):
-
-    def __init__(self, resolver):
-        self._resolver = resolver
-
-    def build_static_endpoints(self, service_names):
-        """Build a set of static endpoints.
-
-        :param service_names: The name of services to build.
-            These are the service names they are supported by
-            boto2.  They also must use the names that boto2
-            uses, not boto3, e.g "ec2containerservice" and not "ecs".
-
-        :return: A dict consisting of::
-            {"service": {"region": "full.host.name"}}
-
-        """
-        static_endpoints = {}
-        for name in service_names:
-            endpoints_for_service = self._build_endpoints_for_service(name)
-            if endpoints_for_service:
-                # It's possible that when we try to build endpoints for services
-                # we get an empty hash.  In that case we don't bother adding
-                # it to the final list of static endpoints.
-                static_endpoints[name] = endpoints_for_service
-        self._deal_with_special_cases(static_endpoints)
-        return static_endpoints
-
-    def _build_endpoints_for_service(self, service_name):
-        # Given a service name, 'ec2', build a dict of
-        # 'region' -> 'hostname'
-        if self._resolver.is_global_service(service_name):
-            return self._special_case_global_service(service_name)
-        endpoints = {}
-        for region_name in self._resolver.regions_for_service(service_name):
-            endpoints[region_name] = self._resolver.get_hostname(service_name,
-                                                                  region_name)
-        return endpoints
-
-    def _special_case_global_service(self, service_name):
-        # In boto2, an entry for each known region is added with the same
-        # partition wide endpoint for every partition the service is available
-        # in.  This method implements this special cased behavior.
-        endpoints = {}
-        for partition in self._resolver.partitions_for_service(service_name):
-            region_names = self._resolver.regions_for_partition(
-                partition)
-            for region_name in region_names:
-                endpoints[region_name] = self._resolver.get_hostname(
-                    service_name, region_name)
-        return endpoints
-
-    def _deal_with_special_cases(self, static_endpoints):
-        # I'm not sure why we do this, but cloudsearchdomain endpoints
-        # use the exact same set of endpoints as cloudsearch.
-        if 'cloudsearch' in static_endpoints:
-            static_endpoints['cloudsearchdomain'] = static_endpoints['cloudsearch']
+def filter_services(endpoints):
```

Comment:
Curious here and just throwing it out there. What would happen if we did not filter the services and just used the exact botocore file? It may be a little strange to have unsupported services in the file but may be less error prone if we can use a straight copy and paste of the file from botocore. What are your thoughts?